notebooks/Emukit-Categorical-with-Tensorflow.ipynb

```diff
@@ -0,0 +1,326 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Categorical variables: optimizing neural network with TensorFlow and Emukit"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook is going to illustrate how to use categorical variables with Emukit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### General imports\n",
+    "%matplotlib inline\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Necessary imports\n",
+    "import tensorflow as tf\n",
+    "import GPy\n",
+    "\n",
+    "from emukit.core import ContinuousParameter, ParameterSpace, CategoricalParameter, OneHotEncoding\n",
+    "from emukit.model_wrappers import GPyModelWrapper\n",
+    "from emukit.bayesian_optimization.loops import BayesianOptimizationLoop"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For this example we are going to optimize a very simple neural network built with TensorFlow. We will reuse the very first example of [how to use TensorFlow](https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/_index.ipynb) with number of epochs reduced to 1 in the interest of saving time. We are going to consider two hyper parameters that are going to be subjects of optimization:\n",
+    "* Type of optimizer. We are going to choose from Adam, AdaGrad and SGD. This is clearly a categorical parameter.\n",
+    "* Dropout rate, ranging between 0.05 and 0.95. This parameter is continuous.  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here is the definition of the function we are going to optimize:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mnist = tf.keras.datasets.mnist\n",
+    "\n",
+    "(x_train, y_train),(x_test, y_test) = mnist.load_data()\n",
+    "x_train, x_test = x_train / 255.0, x_test / 255.0 \n",
+    "\n",
+    "def eval_model(dropout_rate, optimizer):\n",
+    "    model = tf.keras.models.Sequential([\n",
+    "      tf.keras.layers.Flatten(),\n",
+    "      tf.keras.layers.Dense(512, activation=tf.nn.relu),\n",
+    "      tf.keras.layers.Dropout(dropout_rate),\n",
+    "      tf.keras.layers.Dense(10, activation=tf.nn.softmax)\n",
+    "    ])\n",
+    "    model.compile(optimizer=optimizer,\n",
+    "                  loss='sparse_categorical_crossentropy',\n",
+    "                  metrics=['accuracy'])\n",
+    "\n",
+    "    model.fit(x_train, y_train, epochs=1)\n",
+    "    loss, accuracy = model.evaluate(x_test, y_test)\n",
+    "    \n",
+    "    return accuracy"
```

Comment:
I think we should be returning negative accuracy because emukit minimises? I'm not 100% sure though

Reply:
You are correct, did that in emukit_friendly function